src/c2goto/library/string.c

```diff
@@ -26,9 +26,21 @@
 char *strcpy(char *dst, const char *src)
 {
 __ESBMC_HIDE:;
-  char *cp = dst;
-  while ((*cp++ = *src++))
-    ;
+  // Ensure src pointer is non-null
+  if (src == NULL)
+    return NULL;
```

Comment:
Should return the destination ptr

Reply:
https://en.cppreference.com/w/c/string/byte/strcpy
> The behavior is undefined if either dest is not a pointer to a character array or src is not a pointer to a null-terminated byte string.

Still, the whole point of esbmc is to find UB, so we should assert here.